### DISCUSSION TITLE
Unknown arg when using connectOrCreate, but arg exists

### DISCUSSION THREAD

**Discussion by leo-petrucci** [prisma/prisma#8968] [Q&A] [answered]
I'm working on a fairly complex query, but despite following the `connectOrCreate` docs as close as possible I can't seem to get it working.

My schema looks like this: 
```prisma
model Report {
  id         String   @id @default(cuid())
  createdAt  DateTime @default(now())
  user       User     @relation(fields: [userId], references: [id])
  userId     String
  questions  Question[]
  sensors    Sensor[]
}

model Question {
  id         String   @id @default(cuid())
  question   String   // Actual text of the question
  type       String   // Expected type of answer
  reports    Report[]  
  @@unique([id, question])
}

model User {
  id            String    @id @default(cuid())
  email         String?   @unique
  reports       Report[]
  questions     Question[]
}
```

There's no errors and the schema deploys correctly.

But if I try to run the following `create` query:
```ts
    await prisma.report.create({
      data: {
        id,
        createdAt: new Date(createdAt),
        questions: {
          connectOrCreate: questions.map(({ id, question, type }: any) => ({
            where: {
              id: id,
            },
            create: {
              question,
              type,
            },
          })),
        },
        user: {
          connect: {
            id: userId,
          },
        },
      },
    });
```

I get the following error:
```
PrismaClientValidationError: 
Invalid `prisma.report.create()` invocation:

{
  data: {
    id: 'ckr3c5n7j0021b0u9h7arixv9',
    createdAt: new Date('2021-08-01T20:06:16.606Z'),
    questions: {
    ~~~~~~~~~
      connectOrCreate: [
        {
          where: {
            id: 'ckr3c5n7j0021b0u9h7arixv0'
          },
          create: {
            question: 'Did you drink coffee today?',
            type: undefined
          }
        }
      ]
    },
    userId: 'ckr3c5n6s0000b0u9acqs51eg',
    user: {
    ~~~~
      connect: {
        id: 'ckr3c5n6s0000b0u9acqs51eg'
      }
    },
    sensors: {
      create: {
        type: 'geolocation',
        data: '{}'
      }
    }
  }
}

Unknown arg `questions` in data.questions for type ReportUncheckedCreateInput. Available args:     

type ReportUncheckedCreateInput {
  id?: String
  createdAt?: DateTime
  userId: String
}
Unknown arg `user` in data.user for type ReportUncheckedCreateInput. Did you mean `userId`? Available args:
type ReportUncheckedCreateInput {
  id?: String
  createdAt?: DateTime
  userId: String
}
```

A couple of problems there:
- Typescript doesn't allow me to remove the `user.connect` object as its required
- `questions` just doesn't seem to be in any of the `create` typings, which is really confusing.

What am I doing wrong here? I've deleted the prisma client, regenerated it but the error persists. It's like it's not seeing my `question` and `user` types at all.

**Comment by ryands17** [comment]
@creativiii 👋 

As per your error, you are adding both `userId` and `user: { connect: { id: '' } }` which is not possible. You can only use `user: { connect` in this case.

**Reply by leo-petrucci** [reply]
Oops, looks like I left that in. I tried removing `userId` but it still didn't work. I think I figured out the issue though, will post in a second.

**Comment by leo-petrucci** [comment, marked as answer]
Looks like the error was a red herring. The problem was caused by:
```
type: undefined
```
Which I had not noticed.

I'm not entirely sure why that being undefined causes both other connects to break, but once fixed they both worked.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
